api/envoy/api/v2/core/protocol.proto

```diff
@@ -108,6 +111,9 @@ message Http1ProtocolOptions {
       // Note that while this results in most headers following conventional casing, certain headers
       // are not covered. For example, the "TE" header will be formatted as "Te".
       ProperCaseWords proper_case_words = 1;
+
+      // Preserves the case of the header keys
```

Comment:
If you do make the changes I have suggested elsewhere, it should be noted here on the config that this _only_ applies to incoming request headers, not headers set programmatically or via other means. `proper_case_words` applies on all headers, but `preserved_case_words` is only for incoming headers.